include/xtypes/ArrayType.hpp

```diff
@@ -90,6 +90,20 @@ class ArrayType : public CollectionType
     /// \returns The dimension.
     uint32_t dimension() const { return dimension_; }
 
+    std::vector<uint32_t> dimensions() const
+    {
+        std::vector<uint32_t> dimensions_;
+        dimensions_.push_back(dimension_);
+        DynamicType::Ptr inner(*content_);
+        while(inner->kind() == TypeKind::ARRAY_TYPE)
+        {
+            const ArrayType& inner_array = static_cast<const ArrayType&>(*inner);
+            dimensions_.push_back(inner_array.dimension_);
+            inner = *inner_array.content_;
+        }
+        return dimensions_;
+    }
+
```

Comment:
This logic doesn't belong here. The idea of this implementation of `ArrayType` is that it only knows its dimension.
For multidimensional arrays, you must think in arrays of arrays, so this method doesn't have any sense with that in mind.